include/pmacc/nvidia/atomic.hpp

```diff
@@ -23,12 +23,18 @@
 
 
 #include "pmacc/types.hpp"
-#if( PMACC_CUDA_ENABLED == 1 )
+#include "pmacc/memory/Array.hpp"
+#if( BOOST_LANG_CUDA || BOOST_COMP_HIP)
 #   include "pmacc/nvidia/warp.hpp"
 #endif
```

Comment:
I am unaware of reasoning, so perhaps a stupid question. Ins't it better to put such guards inside this file and include it always?